xapian-core/weight/tfidfweight.cc

```diff
@@ -353,12 +357,50 @@ TfIdfWeight::get_wtn(double wt, wt_norm wt_normalization) const
     return wt;
 }
 
+static inline void
+parameter_error(const char* message)
+{
+    Xapian::Weight::Internal::parameter_error(message, "tfidf");
+}
+
 TfIdfWeight *
 TfIdfWeight::create_from_parameters(const char * p) const
 {
     if (*p == '\0')
 	return new Xapian::TfIdfWeight();
-    return new Xapian::TfIdfWeight(p);
+    string wdf_normalisation, idf_normalisation, wt_normalisation;
+    wdf_norm wdf_normalisation_ = wdf_norm::NONE;
+    idf_norm idf_normalisation_ = idf_norm::TFIDF;
+    wt_norm wt_normalisation_ = wt_norm::NONE;
```

Comment:
Better to declare these 3 below at the point you actually have the correct value to assign to them.  As this code is currently written the reader has to read it carefully to see that the values set here are never actually used.

Reply:
Updated